flow/activities/flowable.go

```diff
@@ -692,10 +686,10 @@ func (a *FlowableActivity) replicateQRepPartition(ctx context.Context,
 		logger.Info("no records to push for partition " + partition.PartitionId)
```

Comment:
Ideally `dstConn.SyncQRepRecords(ctx, config, partition, stream)` would also run inside errgroup for pg situation, causing either sync or pull failing to cancel the other

Reply:
I'm not sure if sync should fail if pull fails, if say there are rows currently being pushed, then we should finish pushing them and then get stuck on pull

Reply:
In that case you'd put sync in the errGroup & pass it the non-errctx, so that if sync fails pull stops, removing the need for `pullCancel`

Reply:
Looking at it now, I'm not sure what the point of pullCtx is, we only use it to get the source connector. I think it was relevant back when ctx was part of the connector struct and was being used by all the methods

Reply:
What is pullCancel cancelling exactly

Reply:
Yes, seems like a bug tho, because the idea is that if sync finishes early then `pullCancel()` will cancel the errgroup